docs: document account storeStateStrategy (#9557)
better-auth/better-auth: docs/content/docs/reference/options.mdx

* `"cookie"`: Store the state payload in an encrypted cookie. This keeps OAuth state stateless and avoids a database write during the start of the flow.
* `"database"`: Store the state payload in the verification storage/table, set a signed state cookie when starting the flow, and validate the stored state against the signed cookie sent by the browser on the OAuth callback request.

> Defaults to `"database"` when a database is configured and `"cookie"` for database-less/stateless setups. If you only provide `secondaryStorage`, set this to `"database"` to store OAuth state there instead of in the cookie.

Reviewer: suggestion: I think this could be clearer about 'database' vs 'secondaryStorage'.

I'm assuming (please check!) the default is `"database"` when a 'full' (what's the standard name for a 'non-secondary' database? 'main'? 'full'?) database is configued - but `"cookie"` when it isn't (regardless whether its a 'full' or 'secondaryStorage' database).

So perhaps just clarifying:
```suggestion:
When using only `secondaryStorage`, this will default to `"cookie"`. Set this to `"database"` to store OAuth state there.
```
Actually - I think this points to something confusing. I can't really see why you'd want to use "cookie" here now you can use `secondaryStorage`. I think the default when you have a 'full' database or `secondaryStorage` should _already_ be `"database"`? I found it confusing that I had to set this value - I expected this to just be _how it worked_.

Reviewer 2: this could arguably come down to a code-level issue rather than purely a docs clarification, I recommend opening an issue.
